### PR TITLE
JSONP transport: use jsonp envelope

### PR DESCRIPTION
**Pull request by SimonWoolf**
Fixes https://github.com/ably/ably-js/issues/113 .

Corresponding PR for realtime: https://github.com/ably/realtime/pull/280/files

### CODE REVIEW

File: browser/lib/transport/jsonptransport.js [hunk 1]
```diff
@@ -87,7 +88,17 @@ var JSONPTransport = (function() {
 		};
 
 		_[id] = function(message) {
-			self.complete(null, message);
```
Comment (paddybyers) [inline browser/lib/transport/jsonptransport.js]: Need to insert

````
var response = message.response;
if(!response) {
  callback(new ErrorInfo('Invalid server response: no envelope detected', 50000, 500);
  return;
}
````

File: browser/lib/transport/jsonptransport.js [hunk 2]
```diff
@@ -87,7 +88,17 @@ var JSONPTransport = (function() {
...
+			var successResponse = (message.statusCode < 400);
+			if(successResponse) {
+				self.complete(null, message.response);
+				return;
+			}
+			var err = message.response.error;
+			if(!err) {
```
Comment (paddybyers) [inline browser/lib/transport/jsonptransport.js]:
````
var err = response.error || new ErrorInfo(' .... ', 50000, message.statusCode);
````

**Comment by SimonWoolf** [comment]
done

**Comment by paddybyers** [comment]
Thanks

**Pull request merged**